app/src/main/java/com/niesens/morsetrainer/MainActivity.java

```diff
@@ -145,7 +146,7 @@ private void startTrainer() {
         }
         button_startStop.setText(R.string.trainingStopText);
         final SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
-        trainer = new Trainer(morsePlayer, textSpeaker, wordList, getWordTrainTimesPreference(sharedPreferences), toggleButton_speakFirst.isChecked());
+        trainer = new Trainer(morsePlayer, textSpeaker, wordList, getWordTrainTimesPreference(sharedPreferences), toggleButton_speakFirst.isChecked(), getVocalizePreference(sharedPreferences));
         trainer.execute();
     }
 
```

Comment:
"answer_vocalize" needs to be added to onSharedPreferenceChanged so changes take affect immediately without having to start/stop.